src/tree/AzureAccountTreeItem.ts

```diff
@@ -3,15 +3,77 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AzureAccountTreeItemBase, ISubscriptionContext } from 'vscode-azureextensionui';
+import { pathExists } from 'fs-extra';
+import { join } from 'path';
+import { Disposable, workspace } from 'vscode';
+import { AzExtTreeItem, AzureAccountTreeItemBase, callWithTelemetryAndErrorHandling, IActionContext, ISubscriptionContext } from 'vscode-azureextensionui';
+import { apiSubpathSetting, appArtifactSubpathSetting, appSubpathSetting, configFileName, outputSubpathSetting } from '../constants';
+import { ext } from '../extensionVariables';
+import { createRefreshFileWatcher } from './localProject/createRefreshFileWatcher';
+import { LocalProjectTreeItem } from './localProject/LocalProjectTreeItem';
 import { SubscriptionTreeItem } from './SubscriptionTreeItem';
 
-export class AzureAccountTreeItem extends AzureAccountTreeItemBase {
+export class AzureAccountTreeItemWithProject extends AzureAccountTreeItemBase {
+    private _projectDisposables: Disposable[] = [];
+
     public constructor(testAccount?: {}) {
         super(undefined, testAccount);
+        this.disposables.push(workspace.onDidChangeWorkspaceFolders(async () => {
+            await callWithTelemetryAndErrorHandling('AzureAccountTreeItemWithProject.onDidChangeWorkspaceFolders', async (context: IActionContext) => {
+                context.errorHandling.suppressDisplay = true;
+                context.telemetry.suppressIfSuccessful = true;
+                await this.refresh(context);
+            });
+        }));
+        this.disposables.push(workspace.onDidChangeConfiguration(async event => {
+            await callWithTelemetryAndErrorHandling('AzureAccountTreeItemWithProject.onDidChangeConfiguration', async (context: IActionContext) => {
+                context.errorHandling.suppressDisplay = true;
+                context.telemetry.suppressIfSuccessful = true;
+                const settings: string[] = [appSubpathSetting, apiSubpathSetting, appArtifactSubpathSetting, outputSubpathSetting];
+                if (settings.some(s => event.affectsConfiguration(`${ext.prefix}.${s}`))) {
+                    await this.refresh(context);
+                }
+            });
+        }));
```

Comment:
I'm not sure if this is necessary - will any tree items need to update when these settings change?